test/integration/retryable-reads/retryable_reads.spec.test.js

```diff
@@ -28,3 +37,90 @@ describe('Retryable Reads', function () {
     );
   });
 });
+
+// These tests are skipped because the driver 1) executes a ping when connecting to
+// an authenticated server and 2) command monitoring is at the connection level so
+// when the handshake fails no command started event is emitted.
+const SKIP = [
+  'find succeeds after retryable handshake network error',
+  'find succeeds after retryable handshake network error (ShutdownInProgress)'
+];
+
+describe('Retryable Reads (unified)', function () {
+  runUnifiedSuite(loadSpecTests(path.join('retryable-reads', 'unified')), SKIP);
+});
+
+describe('Retryable Reads Spec Manual Tests', function () {
+  const dbName = 'retryable-handshake-tests';
+  const collName = 'coll';
+  const docs = [
+    { _id: 1, x: 11 },
+    { _id: 2, x: 22 },
+    { _id: 3, x: 33 }
+  ];
+  let client;
+  let db;
+  let coll;
+
+  beforeEach(async function () {
+    if (
+      semver.lt(this.configuration.buildInfo.version, '4.2.0') ||
+      !VALID_TOPOLOGIES.includes(this.configuration.topologyType) ||
+      !this.configuration.options.auth ||
+      !!process.env.SERVERLESS
+    ) {
+      this.currentTest.skipReason =
+        'Retryable reads tests requires authenticated MongoDB 4.2 and higher and no standalone';
+      this.skip();
+    }
+    client = this.configuration.newClient({});
+    db = client.db(dbName);
+    coll = db.collection(collName);
+    await client.connect();
+    await coll.insertMany(docs);
+  });
+
+  afterEach(async function () {
+    if (db) {
+      await db.admin().command({
+        configureFailPoint: 'failCommand',
+        mode: 'off'
+      });
+      await coll.drop();
+      await client.close();
+    }
+  });
+
+  context('when the handshake fails with a network error', function () {
+    // Manual implementation for: 'find succeeds after retryable handshake network error'
+    it('retries the read', async function () {
+      await db.admin().command({
+        configureFailPoint: 'failCommand',
+        mode: { times: 2 },
+        data: {
+          failCommands: ['saslContinue', 'ping'],
+          closeConnection: true
+        }
+      });
+      const doc = await coll.findOne({ _id: 2 });
+      expect(doc).to.deep.equal(docs[1]);
+    });
+  });
+
+  context('when the handshake fails with shutdown in progress', function () {
+    // Manual implementation for:
+    // 'find succeeds after retryable handshake network error (ShutdownInProgress)'
+    it('retries the read', async function () {
+      await db.admin().command({
+        configureFailPoint: 'failCommand',
+        mode: { times: 2 },
+        data: {
+          failCommands: ['saslContinue', 'ping'],
+          errorCode: 91 // ShutdownInProgress
+        }
+      });
+      const documents = await coll.find().toArray();
```

Comment:
-> `find({ _id: 2 })` for consistency with the original test (and update the assertion accordingly)

Reply:
updated this